platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/tipset/Tipset.java

```diff
@@ -0,0 +1,254 @@
+
+    /**
+     * <p>
+     * Merge a list of tipsets together.
+     * </p>
+     *
+     * <p>
+     * The resulting tipset will contain the union of the node IDs of all source tipsets. The generation for each node
+     * ID will be equal to the maximum generation found for that node ID from all source tipsets. If a tipset does not
+     * contain a generation for a node ID, the generation for that node ID is considered to be 0.
+     * </p>
+     *
+     * @param tipsets the tipsets to merge, must be non-empty, tipsets must be constructed from the same address book or
+     *                else this method has undefined behavior
+     * @return a new tipset
+     */
+    public static @NonNull Tipset merge(@NonNull final List<Tipset> tipsets) {
+        Objects.requireNonNull(tipsets, "tipsets must not be null");
+        if (tipsets.isEmpty()) {
+            throw new IllegalArgumentException("Cannot merge an empty list of tipsets");
+        }
+
+        final int length = tipsets.get(0).tips.length;
+        final Tipset newTipset = buildEmptyTipset(tipsets.get(0));
+
+        for (int index = 0; index < length; index++) {
+            long max = 0;
+            for (final Tipset tipSet : tipsets) {
+                max = Math.max(max, tipSet.tips[index]);
+            }
+            newTipset.tips[index] = max;
+        }
+
+        return newTipset;
+    }
+
+    /**
+     * Get the tip generation for a given node ID.
+     *
+     * @param nodeId the node ID in question
+     * @return the tip generation for the node ID
+     */
+    public long getTipGenerationForNodeId(final long nodeId) {
+        return tips[nodeIdToIndex.applyAsInt(nodeId)];
+    }
+
+    /**
+     * Get the tip generation for a given node index. Returns 0 if this node has not been added to the tipset.
+     *
+     * @param index the index of the node in question
+     * @return the tip generation for the node index
+     */
+    public long getTipGenerationForNodeIndex(final int index) { // TODO test
+        return tips[index];
+    }
+
+    /**
+     * Get the number of tips currently being tracked.
+     *
+     * @return the number of tips
+     */
+    public int size() {
+        return tips.length;
+    }
+
+    /**
+     * Advance a single tip within the tipset.
+     *
+     * @return this object
+     */
+    public @NonNull Tipset advance(final long creatorId, final long generation) {
+        final int index = nodeIdToIndex.applyAsInt(creatorId);
+        tips[index] = Math.max(tips[index], generation);
+        return this;
+    }
+
+    /**
+     * <p>
+     * Get the number of tip advancements, weighted using the provided function, between this tipset and another
+     * tipset.
+     * </p>
+     *
+     * <p>
+     * A tip advancement is defined as an increase in the tip generation for a node ID. The exception to this rule is
+     * that an increase in generation for the target node ID is never counted as a tip advancement. The tip advancement
```

Comment:
`is never counted as a tip advancement because self-advancement does nothing to help consensus.`  <== simply adding the comment at L173 below to this sentence.